pkg/skaffold/deploy/kubectl/cli.go

```diff
@@ -26,25 +26,29 @@ import (
 
 	"github.com/sirupsen/logrus"
 
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
 	pkgkubectl "github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubectl"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
 )
 
-var WaitDeletion = struct {
-	MaxRetry int
-	Delay    time.Duration
-}{
-	MaxRetry: 30,
-	Delay:    2 * time.Second,
-}
-
 // CLI holds parameters to run kubectl.
 type CLI struct {
 	*pkgkubectl.CLI
 	Flags latest.KubectlFlags
 
-	ForceDeploy   bool
-	previousApply ManifestList
+	forceDeploy      bool
+	waitForDeletions config.WaitForDeletions
+	previousApply    ManifestList
+}
+
+func NewCLI(runCtx *runcontext.RunContext, flags latest.KubectlFlags) CLI {
```

Comment:
I'd prefer that this not take the RunContext and instead pass in the configuration values explicitly: this CLI is a utility library and using the RunContext ties it to the Skaffold configuration.  We're now mixing policy with the mechanism.

Reply:
Do you mind if I do it in another PR? I would then do the same for kubectl.NewFromRunContext

Reply:
Oops, I thought this was in `pkg/skaffold/kubectl`, not `pkg/skaffold/deploy/kubectl`.  I retract my comment.